sdcm/cluster.py

```diff
@@ -1683,35 +1683,51 @@ def get_scylla_version(self):
                     else:
                         self.log.error("Unknown ScyllaDB version")
 
-    def cfstat_reached_threshold(self, key, threshold, keyspace):
+    def run_cqlsh(self, node, cql_cmd, timeout=60, verbose=True):
+        cqlsh_out = node.remoter.run('cqlsh -e "{}" {}'.format(cql_cmd, node.private_ip_address), timeout=timeout, verbose=verbose)
+        return cqlsh_out.stdout
+
+    def get_test_keyspaces(self):
+        out = self.run_cqlsh(node=self.nodes[0], cql_cmd='select keyspace_name from system_schema.keyspaces')
+        return [ks.strip() for ks in out.split('\n')[3:-3] if 'system' not in ks]
+
+    def cfstat_reached_threshold(self, key, threshold, keyspaces=None):
         """
         Find whether a certain cfstat key in all nodes reached a certain value.
 
         :param key: cfstat key, example, 'Space used (total)'.
         :param threshold: threshold value for cfstats key. Example, 2432043080.
-        :param keyspace: keyspace name or table full name(example: 'keyspace1.standard1')
+        :param keyspace: keyspace name or table full name(example: 'keyspace1.standard1'). If keyspaces is None,
+                         receive all
         :return: Whether all nodes reached that threshold or not.
         """
-        tcpdump = self.params.get('tcpdump')
-        cfstats = [node.get_cfstats(keyspace, tcpdump)[key] for node in self.nodes]
-        self.log.debug("Current cfstats: %s" % cfstats)
+        if not keyspaces:
+            keyspaces = self.get_test_keyspaces()
+            # out = self.run_cqlsh(node=self.nodes[0], cql_cmd='select keyspace_name from system_schema.keyspaces')
+            # keyspaces = [ks.strip() for ks in out.split('\n')[3:-3] if 'system' not in ks]
+
+        cfstats = []
+        for keyspace_name in keyspaces:
+            ks_stats = [node.get_cfstats(keyspace_name)[key] for node in self.nodes]
+            cfstats.append(min(ks_stats))
+            self.log.debug("Current cfstats for %s keyspace: %s" % (keyspace_name, ks_stats))
```

Comment:
Let's print in this log also the threshold we are waiting for, so it'll east to detect issues.